tests/scripts/components-build-system.sh

```diff
@@ -36,3 +53,83 @@ component_test_tf_psa_crypto_cmake_as_package_install () {
     make
     ./cmake_package_install
 }
+
+component_tf_psa_crypto_build_custom_config_file () {
+    # Make a copy of config file
+    cp "$CRYPTO_CONFIG_H" include/psa/crypto_config_default.h
+
+    # Build once to get the generated files (for which need an intact config file)
+    msg "build: cmake out-of-source with default config file"
+    cd "$OUT_OF_SOURCE_DIR"
+    cmake "$TF_PSA_CRYPTO_ROOT_DIR"
+    make
+
+    msg "build: cmake out-of-source with -DMBEDTLS_CONFIG_FILE"
+    cd "$TF_PSA_CRYPTO_ROOT_DIR"
+    cp "$CRYPTO_CONFIG_H" "$OUT_OF_SOURCE_DIR/crypto_config_custom.h"
+    cd "$OUT_OF_SOURCE_DIR"
+    # Force a build failure if the default crypto_config.h header file is included
+    # in the build instead of the one we pass to CMake on the command line.
+    echo '#error "cmake -DMBEDTLS_CONFIG_FILE is not working."' > "$TF_PSA_CRYPTO_ROOT_DIR/$CRYPTO_CONFIG_H"
+    cmake -DGEN_FILES=OFF -DTF_PSA_CRYPTO_CONFIG_FILE=crypto_config_custom.h "$TF_PSA_CRYPTO_ROOT_DIR"
+    make
+
+    msg "build: cmake out-of-source with -DTF_PSA_CRYPTO_CONFIG_FILE + -DTF_PSA_CRYPTO_USER_CONFIG_FILE"
+    # In the user config, disable one feature (for simplicity, pick a feature
+    # that nothing else depends on).
+    echo '#undef MBEDTLS_NIST_KW_C' >crypto_config_user.h
+
+    cmake -DGEN_FILES=OFF -DTF_PSA_CRYPTO_CONFIG_FILE=crypto_config_custom.h \
```

Comment:
```suggestion
    grep -q mbedtls_nist_kw_init drivers/builtin/libbuiltin.a
    cmake -DGEN_FILES=OFF -DTF_PSA_CRYPTO_CONFIG_FILE=crypto_config_custom.h \
```
to check the symbol `mbedtls_nist_kw_init` is actually in the archive before trying to remove it.